src/analysis/calculus/deriv.lean

```diff
@@ -980,12 +1009,14 @@ begin
   exact eq₁.tri eq₃
 end
 
-theorem has_fderiv_within_at.comp {g : F → G} {g' : F →L[𝕜] G}
-  (hg : has_fderiv_within_at g g' (f '' s) (f x))
-  (hf : has_fderiv_within_at f f' s x) :
+theorem has_fderiv_within_at.comp {g : F → G} {g' : F →L[𝕜] G} {t : set F}
+  (hg : has_fderiv_within_at g g' t (f x)) (hf : has_fderiv_within_at f f' s x) (hst : s ⊆ f ⁻¹' t) :
   has_fderiv_within_at (g ∘ f) (g'.comp f') s x :=
-(has_fderiv_at_filter.mono hg
-  hf.continuous_within_at.tendsto_nhds_within_image).comp x hf
+begin
```

Comment:
Maybe 
```lean
  apply has_fderiv_at_filter.comp _ (has_fderiv_at_filter.mono hg _) hf,
  calc map f (nhds_within x s) 
      ≤ nhds_within (f x) (f '' s) : hf.continuous_within_at.tendsto_nhds_within_image
  ... ≤ nhds_within (f x) t        : nhds_within_mono _ (image_subset_iff.mpr hst)
```
(when I see `le_trans` I always think calc block).

Reply:
Yes, it is clearer like that.